explainer.md

```diff
@@ -22,118 +71,260 @@ place associated with the application (such as the shelf or home screen).
 
 Advantages of using the badging API over notifications:
 
-* Can be used for much higher frequency events than notifications, because each
-  new event does not disrupt the user.
-* There is no need to request permission to use the badging API, since it is
-  much less invasive than a notification.
+* Can be used for much higher frequency events than notifications, because each new event does not disrupt the user.
+* There is no need to request permission to use the badging API, since it is much less invasive than a notification.
 
-(Typically, sites will want to use both APIs together: notifications for
-high-importance events such as new direct messages or incoming calls, and badges
-for all new messages including group chats not directly addressed to the user.)
+(Typically, sites will want to use both APIs together: notifications for high-importance events such as new direct messages or incoming calls, and badges for all new messages including group chats not directly addressed to the user.)
 
 ## API proposal
 
 ### The model
 
-There is a single global badge associated with each Web
-application (as defined in [Web app
-manifest](https://www.w3.org/TR/appmanifest/)). At any time, the badge is set with:
+1. A Badge is associated with a [scope](https://www.w3.org/TR/appmanifest/#navigation-scope).
+2. Documents are badged with the most specific badge for their url (i.e. prefer a badge for `/page/1` to a badge for `/page/` when on the url `/page/1?foo=7`).
+3. If no scope is specified the scope is the current origin (equivalent to setting the scope to `/`).
+4. For [installed applications](https://www.w3.org/TR/appmanifest/#installable-web-applications), a user agent **MAY** display the badge with the most specific scope still encompassing the [navigation scope](https://www.w3.org/TR/appmanifest/#navigation-scope) of the application in an [OS specific context](#OS-Specific-Contexts).
 
-* Nothing (the badge is "cleared"), or
+At any time, the badge for a specific scope may be:
+
+* Nothing. Apply the badge for a less specific scope, or, if there is no matching badge, display nothing.
 * A "flag" indicating the presence of a badge with no contents, or
 * A positive integer.
 
-The model does not allow a badge that is a negative integer, or the integer value 0
-(setting the badge to 0 is equivalent to clearing the badge).
+The model does not allow a badge to be a negative integer, or the integer value 0 (setting the badge to 0 is equivalent to clearing the badge).
 
 ### The API
 
 The `Badge` interface is a member object on
-[`Window`](https://html.spec.whatwg.org/#the-window-object). It contains two methods:
+[`Window`](https://html.spec.whatwg.org/#the-window-object). It contains three methods:
+
+```webidl
+dictionary BadgeOptions {
+  // The scope the badge applies to.
+  // If unspecified this is the origin of the current page.
+  USVString scope;
```

Comment:
Still worried about managing this in the browser, specially across multiple tabs :( I wonder if we should make this simple in that, if the page navigates, the badge needs to be set again. Need to think about this more.

Reply:
I think we should say that the user agent **MAY** clear the badge on navigation but I don't think we should require it.

Sites should always update their badges when they're loaded but after they have been closed it might be useful to keep showing a badge count somewhere (e.g. in the dock or in a bookmark).

I agree that managing this looks pretty daunting though :/

Reply:
We might need to do some prototypes... just to see how sane it is. I know a lot of people at Mozilla don't like SW scopes also (for same reasons around complexity) - so need to have a coherent story/model to show how this will work.

Reply:
Yeah, it's definitely a worry. I wonder if we can avoid going down this rabbit hole?

Do you want to create a new issue for it and see what some people more knowledgeable than me here (almost anyone :P ) think?

Reply:
Yes, that's a good idea... when you file the bug, just describe the problem and say that the problem "feels somewhat similar to a service worker scope, but we completely open to alternatives".

Reply:
Ok... thinking more about this... maybe we don't need scope, and service worker might be the answer here (because it already has scope): What if the management of the badge with written in JS and managed by the service worker? It could then use `BroadcastChannel` to notify other tabs if they need to update the badge?

Reply:
And navigating would clear badge... but could be set again `onload` or whatever.

Reply:
Created #43, shall we move discussion there?